keel-core/src/main/kotlin/com/netflix/spinnaker/keel/pause/ActuationPauser.kt

```diff
@@ -66,59 +69,55 @@ class ActuationPauser(
     log.info("Resuming application $application")
     pausedRepository.resumeApplication(application)
-    publisher.publishEvent(ApplicationActuationResumed(application))
-    resourceRepository.getResourcesByApplication(application)
-      .forEach {
-        // helps a user not be confused by an out of date status from before a pause
-        publisher.publishEvent(ResourceActuationResumed(it))
-      }
+    publisher.publishEvent(ApplicationActuationResumed(application, "Application $application resumed", clock))
   }
 
   fun pauseResource(id: String) {
     log.info("Pausing resource $id")
     pausedRepository.pauseResource(id)
-    publisher.publishEvent(ResourceActuationPaused(resourceRepository.get(id)))
+    publisher.publishEvent(ResourceActuationPaused(resourceRepository.get(id), null, clock.instant()))
   }
 
   fun resumeResource(id: String) {
     log.info("Resuming resource $id")
     pausedRepository.resumeResource(id)
     // helps a user not be confused by an out of date status from before a pause
-    publisher.publishEvent(ResourceActuationResumed(resourceRepository.get(id)))
+    publisher.publishEvent(ResourceActuationResumed(resourceRepository.get(id), clock))
   }
 
   fun pausedApplications(): List<String> =
     pausedRepository.getPausedApplications()
 
-  fun addSyntheticPausedEvents(originalEvents: List<ResourceEvent>, resource: Resource<*>) =
-    originalEvents.toMutableList().also { events ->
-      // For user clarity we add a pause event to the resource history for every pause event from the parent app.
-      // We do this dynamically here so that it applies to all resources in the app, even those added _after_ the
-      // application was paused.
-      val appPausedEvents = resourceRepository
-        .applicationEventHistory(resource.application, events.last().timestamp)
-        .filterIsInstance<ApplicationActuationPaused>()
+  /**
+   * Adds [ApplicationActuationPaused] and [ApplicationActuationResumed] events to the resource event history so that
+   * it reflects actuation pauses/resumes at the application level.
+   */
+  fun addApplicationActuationEvents(originalEvents: List<ResourceEvent>, resource: Resource<*>) =
+    mutableListOf<PersistentEvent>()
+      .also { events ->
+        events.addAll(originalEvents)
+        val oldestResourceEvent = events.last()
+        val relevantAppEvents = resourceRepository
+          .applicationEventHistory(resource.application, oldestResourceEvent.timestamp)
+          .filter { it is ApplicationActuationPaused || it is ApplicationActuationResumed }
 
-      appPausedEvents.forEach { appPaused ->
-        val lastBeforeAppPaused = events.firstOrNull { event ->
-          event.timestamp.isBefore(appPaused.timestamp)
-        }
+        relevantAppEvents.forEach { appEvent ->
+          val lastBeforeAppEvent = events.firstOrNull { event ->
+            event.timestamp.isBefore(appEvent.timestamp)
+          }
```

Comment:
Couldn't we just apply `sortedBy { it.timestamp }` after we have the list rather than having to work out where to insert things?

Reply:
Er... That would make total sense, wouldn't it? 🤦‍♂ Will change.